pkg/sfu/buffer/rtpstats_receiver.go

```diff
@@ -374,55 +358,104 @@ func (r *RTPStatsReceiver) SetRtcpSenderReportData(srData *RTCPSenderReportData)
 			)
 		}
-		r.outOfOrderSenderReportCount++
-		return false
+		return true
 	}
 
-	if r.srNewest != nil {
-		timeSinceLast := srData.NTPTimestamp.Time().Sub(r.srNewest.NTPTimestamp.Time()).Seconds()
-		rtpDiffSinceLast := srDataCopy.RTPTimestampExt - r.srNewest.RTPTimestampExt
-		calculatedClockRateFromLast := float64(rtpDiffSinceLast) / timeSinceLast
-
-		timeSinceFirst := srData.NTPTimestamp.Time().Sub(r.srFirst.NTPTimestamp.Time()).Seconds()
-		rtpDiffSinceFirst := srDataCopy.RTPTimestampExt - r.srFirst.RTPTimestampExt
-		calculatedClockRateFromFirst := float64(rtpDiffSinceFirst) / timeSinceFirst
-
-		if (timeSinceLast > 0.2 && math.Abs(float64(r.params.ClockRate)-calculatedClockRateFromLast) > 0.2*float64(r.params.ClockRate)) ||
-			(timeSinceFirst > 0.2 && math.Abs(float64(r.params.ClockRate)-calculatedClockRateFromFirst) > 0.2*float64(r.params.ClockRate)) {
-			if r.clockSkewCount%100 == 0 {
-				r.logger.Infow(
-					"received sender report, clock skew",
-					"current", &srDataCopy,
-					"timeSinceFirst", timeSinceFirst,
-					"rtpDiffSinceFirst", rtpDiffSinceFirst,
-					"calculatedFirst", calculatedClockRateFromFirst,
-					"timeSinceLast", timeSinceLast,
-					"rtpDiffSinceLast", rtpDiffSinceLast,
-					"calculatedLast", calculatedClockRateFromLast,
-					"count", r.clockSkewCount,
-					"rtpStats", lockedRTPStatsReceiverLogEncoder{r},
-				)
-			}
-			r.clockSkewCount++
+	return false
+}
+
+func (r *RTPStatsReceiver) checkRTPClockSkewForSenderReport(srData *RTCPSenderReportData) {
+	if r.srNewest == nil {
+		return
+	}
+
+	timeSinceLast := srData.NTPTimestamp.Time().Sub(r.srNewest.NTPTimestamp.Time()).Seconds()
+	rtpDiffSinceLast := srData.RTPTimestampExt - r.srNewest.RTPTimestampExt
+	calculatedClockRateFromLast := float64(rtpDiffSinceLast) / timeSinceLast
+
+	timeSinceFirst := srData.NTPTimestamp.Time().Sub(r.srFirst.NTPTimestamp.Time()).Seconds()
+	rtpDiffSinceFirst := srData.RTPTimestampExt - r.srFirst.RTPTimestampExt
+	calculatedClockRateFromFirst := float64(rtpDiffSinceFirst) / timeSinceFirst
+
+	if (timeSinceLast > 0.2 && math.Abs(float64(r.params.ClockRate)-calculatedClockRateFromLast) > 0.2*float64(r.params.ClockRate)) ||
+		(timeSinceFirst > 0.2 && math.Abs(float64(r.params.ClockRate)-calculatedClockRateFromFirst) > 0.2*float64(r.params.ClockRate)) {
+		r.clockSkewCount++
+		if (r.clockSkewCount-1)%100 == 0 {
+			r.logger.Infow(
+				"received sender report, clock skew",
+				"current", srData,
+				"timeSinceFirst", timeSinceFirst,
+				"rtpDiffSinceFirst", rtpDiffSinceFirst,
+				"calculatedFirst", calculatedClockRateFromFirst,
+				"timeSinceLast", timeSinceLast,
+				"rtpDiffSinceLast", rtpDiffSinceLast,
+				"calculatedLast", calculatedClockRateFromLast,
+				"count", r.clockSkewCount,
+				"rtpStats", lockedRTPStatsReceiverLogEncoder{r},
+			)
 		}
 	}
+}
 
+func (r *RTPStatsReceiver) checkRTPClockSkewAgainstMediaPathForSenderReport(srData *RTCPSenderReportData) {
```

Comment:
This is the new function to check sender report RTP time stamp against media path time stamp. This is the main change in this PR. Rest is just breaking up `SetRtcpSenderReportData()` into smaller bits.